contracts/stake/src/contract.rs

```diff
@@ -754,6 +764,82 @@ impl StakingTrait for Staking {
         distribution.distributed_total
     }
 
+    fn update_config(
+        env: Env,
+        lp_token: Option<Address>,
+        min_bond: Option<i128>,
+        min_reward: Option<i128>,
+        manager: Option<Address>,
+        owner: Option<Address>,
+        max_complexity: Option<u32>,
+    ) -> Result<Config, ContractError> {
+        env.storage()
+            .instance()
+            .extend_ttl(INSTANCE_RENEWAL_THRESHOLD, INSTANCE_TARGET_TTL);
+
+        let admin = get_admin_old(&env);
+        admin.require_auth();
+
+        let mut config = get_config(&env);
+
+        if let Some(lp_token) = lp_token {
+            config.lp_token = lp_token;
+        }
+
+        if let Some(min_bond) = min_bond {
+            if min_bond <= 0 {
+                log!(
+                &env,
+                "Stake: initialize: Minimum amount of lp share tokens to bond can not be smaller or equal to 0"
+            );
+                panic_with_error!(&env, ContractError::InvalidMinBond);
+            }
+            config.min_bond = min_bond
```

Comment:
I guess here we can take just unsigned value and cast it. 
Now as I'm thinking we should replace that in during the initialization as well.

Reply:
casting or using unsigned values at initialization would force me to make changes to Config and  storage. The values in state are already using i128 and the test that's checking the migration `upgrade_staking_contract_and_remove_stake_rewards` fails on `unbond_deprecated` because of that change.

I'd say we keep this as is for now, and we later do a follow-up migration for that. We already have a lot of changes in `Stake` contract as is